### ISSUE TITLE
Textual Inversion not working with Prodigy

### ISSUE THREAD

**Issue by Poiuytrezay1** [kohya-ss/sd-scripts#980]
The LR is stuck below 1e-6 when training textual inversion using Prodigy as the optimizer.

**Comment by feffy380** [comment]
Maybe prodigy doesn't react well to how most of the embeddings are overwritten with the original copies after each optimizer step?

**Comment by feffy380** [comment]
I can confirm that zeroing out the gradients of the other tokens before stepping the optimizer prevents DAdaptAdam from exploding. Maybe it'll help with Prodigy too.
The current code forcibly resets the values after stepping the optimizer, which seems wrong to me because we should be preventing unwanted gradient updates, not reversing them after the fact.

```python
# zero out gradients for all tokens we aren't training
for text_encoder, index_no_updates in zip(text_encoders, index_no_updates_list):
  input_embeddings_weight = accelerator.unwrap_model(text_encoder).get_input_embeddings().weight
  input_embeddings_weight.grad[index_no_updates] = 0

optimizer.step()
lr_scheduler.step()
optimizer.zero_grad(set_to_none=True)

# with torch.no_grad():
    # Let's make sure we don't update any embedding weights besides the newly added token
    # for text_encoder, orig_embeds_params, index_no_updates in zip(
    #     text_encoders, orig_embeds_params_list, index_no_updates_list
    # ):
    #     # if full_fp16/bf16, input_embeddings_weight is fp16/bf16, orig_embeds_params is fp32
    #     input_embeddings_weight = accelerator.unwrap_model(text_encoder).get_input_embeddings().weight
    #     input_embeddings_weight[index_no_updates] = orig_embeds_params.to(input_embeddings_weight.dtype)[
    #         index_no_updates
    #     ]

```
